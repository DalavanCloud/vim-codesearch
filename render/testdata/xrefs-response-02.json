{
    "xref_search_response": [
        {
            "status": 0, 
            "estimated_total_type_count": [
                {
                    "count": 2, 
                    "type": "DEFINITION", 
                    "type_id": "DEFINITION"
                }, 
                {
                    "count": 1, 
                    "type": "DECLARATION", 
                    "type_id": "DECLARATION"
                }, 
                {
                    "count": 1, 
                    "type": "REFERENCE", 
                    "type_id": "REFERENCE"
                }
            ], 
            "search_result": [
                {
                    "file": {
                        "name": "src/net/http/http_server_properties_manager.cc", 
                        "package_name": "chromium"
                    }, 
                    "match": [
                        {
                            "line_text": "      &HttpServerPropertiesManager::OnHttpServerPropertiesChanged,", 
                            "line_number": 121, 
                            "type_id": "REFERENCE", 
                            "type": "REFERENCE", 
                            "signature": "kythe://chromium?lang=c%2B%2B?path=src/net/http/http_server_properties_manager.cc#%404398%3A4427"
                        }, 
                        {
                            "line_text": "void HttpServerPropertiesManager::OnHttpServerPropertiesChanged() {", 
                            "line_number": 1284, 
                            "type_id": "DEFINITION", 
                            "type": "DEFINITION", 
                            "signature": "kythe://chromium?lang=c%2B%2B?path=src/net/http/http_server_properties_manager.cc#%4050051%3A50080%40pr86o0tlYJqeE7i0UHuKyF1nqalq5iJTBS2fnnY%2Fq80%3D%23chromium%23src%2Fnet%2Fhttp%2Fhttp_server_properties_manager.cc"
                        }, 
                        {
                            "line_text": "void HttpServerPropertiesManager::OnHttpServerPropertiesChanged() {", 
                            "line_number": 1284, 
                            "type_id": "DEFINITION", 
                            "type": "DEFINITION", 
                            "signature": "kythe://chromium?lang=c%2B%2B?path=src/net/http/http_server_properties_manager.cc#%4050051%3A50080%40pr86o0tlYJqeE7i0UHuKyF1nqalq5iJTBS2fnnY%2Fq80%3D%23chromium%23src%2Fnet%2Fhttp%2Fhttp_server_properties_manager.cc"
                        }
                    ]
                }, 
                {
                    "file": {
                        "name": "src/net/http/http_server_properties_manager.h", 
                        "package_name": "chromium"
                    }, 
                    "match": [
                        {
                            "line_text": "  void OnHttpServerPropertiesChanged();", 
                            "line_number": 204, 
                            "type_id": "DECLARATION", 
                            "type": "DECLARATION", 
                            "signature": "kythe://chromium?lang=c%2B%2B?path=src/net/http/http_server_properties_manager.h#%408771%3A8800%40LGmPjlk8Ba2AD%2FyGPSqmF8foVgHxZk188FAxPOuLcHY%3D%23chromium%23src%2Fnet%2Fhttp%2Fhttp_server_properties_manager.h"
                        }
                    ]
                }
            ], 
            "from_kythe": true, 
            "grok_total_number_of_results": 0
        }
    ]
}
