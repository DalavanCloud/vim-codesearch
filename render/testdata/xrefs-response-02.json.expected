This file contains the rendered output per line and its associated metadata.

Lines that begin with a line number contains (after the '|') the contents that
will be inserted into the vim buffer at that line. The immediately following
line contains an object representing associated metadata.
-------------------------------------------------------------------------------

001|^Cat{Definition:
   |{}
002|}Cat_  ^F{src/net/http/http_server_properties_manager.cc}F_
   |{'j': ('src/net/http/http_server_properties_manager.cc', 0, 7)}
003|    1284:  ^>{void HttpServerPropertiesManager::OnHttpServerPropertiesChanged() {}>_
   |{'j': ('src/net/http/http_server_properties_manager.cc', 1283, 11)}
004|    1284:  ^>{void HttpServerPropertiesManager::OnHttpServerPropertiesChanged() {}>_
   |{'j': ('src/net/http/http_server_properties_manager.cc', 1283, 11)}
005|
   |{}
006|^Cat{Declaration:
   |{}
007|}Cat_  ^F{src/net/http/http_server_properties_manager.h}F_
   |{'j': ('src/net/http/http_server_properties_manager.h', 0, 7)}
008|    204:  ^>{  void OnHttpServerPropertiesChanged();}>_
   |{'j': ('src/net/http/http_server_properties_manager.h', 203, 10)}
009|
   |{}
010|^Cat{References:
   |{}
011|}Cat_  ^F{src/net/http/http_server_properties_manager.cc}F_
   |{'j': ('src/net/http/http_server_properties_manager.cc', 0, 7)}
012|    121:  ^>{      &HttpServerPropertiesManager::OnHttpServerPropertiesChanged,}>_
   |{'j': ('src/net/http/http_server_properties_manager.cc', 120, 10)}
013|
   |{}
014|
   |{}
